src/drivers/stdio.ts

```
@@ -39,7 +40,50 @@ export function print2(message: any) {
 const AnsiCodes = {
   Reset: "00",
   Dim: "02",
+  Green: "32",
   Yellow: "33",
 } as const;
 
-export const Ansi = mapValues(AnsiCodes, (v) => `\u001b[${v}m`);
+const Ansi = (value: string) => `\u001b[${value}`;
+
+/** Creates an ANSI Select Graphic Rendition code */
+export const AnsiSgr = mapValues(AnsiCodes, (v) => Ansi(v + "m"));
+
+/** Resets the terminal cursor to the beginning of the line */
+export function reset2() {
+  process.stderr.write(Ansi("0G"));
+}
+
+/** Clears the current terminal line */
+export function clear2() {
+  // Replaces text with spaces
+  process.stderr.write(Ansi("2K"));
+  reset2();
+}
+
+const Spin = {
+  items: ["⠇", "⠋", "⠙", "⠸", "⠴", "⠦"],
+  delayMs: 200,
+};
+
+/** Prints a Unicode spinner until a promise resolves */
+export const spinUntil = async <T>(message: string, promise: Promise<T>) => {
+  let isDone = false;
+  let ix = 0;
+  void promise.finally(() => (isDone = true)).catch(() => {});
```

Should we swallow errors here?

---

The errors are thrown on line 77